addons/mail/static/src/messaging/component/follower_subtypes_editor/follower_subtypes_editor.js

```diff
@@ -0,0 +1,82 @@
+odoo.define('mail.messaging.component.FollowerSubtypesEditor', function (require) {
+'use strict';
+
+const components = {
+    FollowerSubtype: require('mail.messaging.component.FollowerSubtype'),
+};
+const useStore = require('mail.messaging.component_hook.useStore');
+
+const Dialog = require('web.OwlDialog');
+
+const { Component, QWeb } = owl;
+
+class FollowerSubtypesEditor extends Component {
+
+    /**
+     * @override
+     */
+    constructor(...args) {
+        super(...args);
+        useStore(props => {
+            const followerSubtypesEditor = this.env.entities.FollowerSubtypesEditor.get(props.localId);
+            const follower = followerSubtypesEditor
+                ? followerSubtypesEditor.follower
+                : undefined;
+            return {
+                follower,
+                followerSubtypes: follower ? follower.subtypes : [],
+                followerSubtypesEditor,
+            };
+        });
+    }
+
+    //--------------------------------------------------------------------------
+    // Public
+    //--------------------------------------------------------------------------
+
+    /**
+     * @returns {mail.messaging.entity.FollowerSubtypesEditor}
+     */
+    get followerSubtypesEditor() {
+        return this.env.entities.FollowerSubtypesEditor.get(this.props.localId);
+    }
+
+    //--------------------------------------------------------------------------
+    // Handlers
+    //--------------------------------------------------------------------------
+
+    /**
+     * Called when clicking on cancel button.
+     *
+     * @private
+     * @param {MouseEvent} ev
+     */
+    _onClickCancel(ev) {
+        this.followerSubtypesEditor.close();
+    }
+
+    /**
+     * Called when clicking on apply button.
+     *
+     * @private
+     * @param {MouseEvent} ev
+     */
+    _onClickApply(ev) {
+        this.followerSubtypesEditor.close();
+        this.followerSubtypesEditor.follower.updateFollowerSubtypes();
+    }
+}
+
+Object.assign(FollowerSubtypesEditor, {
+    components: Object.assign({ Dialog }, components),
+    props: {
+        localId: String,
```

Comment:
(Now it's called follower_subtype_list)
I agree but it's due to how dialogs are developed, currently it instantiates the component like this : 

```xml
<t
    t-component="{{ dialog.entity['constructor'].name }}"
    class="o_Dialog_component"
    t-props="{ localId: dialog.entity.localId }"
    t-ref="component"
/>
```

so `localId` has to be named like that

Reply:
I don't really know well the entity dialog code but it appears FollowerSubtypesEditor/List is a component, and dialog uses an entity, so there has to be a way?

Reply:
Yes there is a way but we need to change how dialog component instantiates props from the entity he received. For the moment he only accepts to use a certain property "localId" of the entity. We could ask the entity to determine the props its component needs but for me it's weird that an entity determines how the component has to be created.

Reply:
This is a consequence from the API to create a UI component in a dialog using entities (dialog entity linked to a custom UI entity): dialog component contains another component and provides a prop that maps to the correct custom UI entity, but this name of prop must be either generic (like `localId` here) or the `.open()` method in dialog manager should provide option to rename this prop.

I think it makes sense for component that directly map an entity to use a generic prop `localId` that identifies this entity, and it's not ambiguous because entity and component make one.